src/ui/components/MapBox/MapBox.js

```diff
@@ -0,0 +1,252 @@
+				const apisTotal = m_props.apis.length * 2;
+
+				ui.d.append("head", "<!-- Load Mapbox Libraries -->");
+
+				m_props.apis.forEach((api, index) => {
+					let script;
+					let link;
+					let arr = [];
+
+					if (api.js) {
+						script = ui.d.createTag("script", {
+							prop: {
+								src: api.js
+							},
+							attr: {
+								"data-mapboxscripts": `js-${index}`
+							}
+						});
+						ui.d.append("head", script);
+						arr.push(script);
+					}
+					if (api.css) {
+						link = ui.d.createTag("link", {
+							prop: {
+								href: api.css,
+								rel: "stylesheet"
+							},
+							attr: {
+								"data-mapboxscripts": `css-${index}`
+							}
+						});
+						ui.d.append("head", link);
+						arr.push(link);
+					}
+
+					arr.forEach(tag => {
+						tag.addEventListener("load", () => {
+							apisLoaded++;
+							if (apisLoaded === apisTotal) {
+								resolve();
+							}
+						});
+					});
+				});
+
+				ui.d.append("head", "<!-- End Mapbox Libraries -->");
+
+			});
+		}
+
+		function setMapBoxAccessToken() {
+			return new Promise((resolve) => {
+				mapboxgl.accessToken = m_props.access_token;
+				resolve();
+			});
+		}
+
+		function setupDOM() {
+			return new Promise((resolve) => {
+				const tagConfig = {
+					class: m_props.css.body,
+					attr: m_props.attr,
+					prop: m_props.prop,
+				};
+
+				m_divTag = ui.d.createTag(m_props.tags.div, tagConfig);
+				self.classList.add(m_props.css.self);
+				ui.d.append(self, m_divTag);
+				resolve();
+			});
+		}
+
+		function getUserLocation() {
+			return new Promise((resolve) => {
+				navigator.geolocation.getCurrentPosition(s => {
+					const lng = s.coords.longitude;
+					const lat = s.coords.latitude;
+					setUserLocation(lng, lat);
+					onMoveEnd(removeWait);
+					jumpTo(lng, lat);
+					addCurrentPositionMarked(lng, lat);
+					resolve();
+				}, () => {
+					geolocationError();
+					resolve();
+				});
+			});
+		}
+
+		function jumpTo(lng, lat) {
+			m_map.jumpTo({
+				center: [lng, lat],
+				zoom: m_props.zoom
+			});
+		}
+
+		function geolocationError(e) {
+			removeWait();
+		}
+
+		function setUserLocation(lat, lng) {
+			m_userLocation = {
+				latitude: lat,
+				longitude: lng
+			};
+		}
+
+		function renderMap(lng, lat) {
+			return new Promise((resolve) => {
+				m_map = new mapboxgl.Map({
+					container: m_props.prop.id,
+					style: 'mapbox://styles/mapbox/streets-v11',
+					zoom: m_zoomDefault,
+				});
+
+				if (m_props.fullscreen) m_map.addControl(new mapboxgl.FullscreenControl());
+				if (m_props.navigation) m_map.addControl(new mapboxgl.NavigationControl());
+
+				if (m_props.search) {
+					m_map.addControl(
+						new MapboxGeocoder({
+							accessToken: mapboxgl.accessToken,
+							mapboxgl: mapboxgl
+						}),
+						"top-left"
+					);
+				}
+
+				m_map.on("load", resolve);
+			});
+		}
+
+		function addCurrentPositionMarked(lng, lat) {
+			const el = ui.d.createTag("div", {
```

Comment:
Tag name "div" must be configurable through **m_props.tags** object

Reply:
Mapbox require a div element. I tested with a custom element but no rendered correctly.

Reply:
Done!